libcst/metadata/expression_context_provider.py

```diff
@@ -155,6 +155,29 @@ def visit_List(self, node: cst.List) -> Optional[bool]:
     def visit_StarredElement(self, node: cst.StarredElement) -> Optional[bool]:
         self.provider.set_metadata(node, self.context)
 
+    def visit_ClassDef(self, node: cst.ClassDef) -> Optional[bool]:
+        node.name.visit(
+            ExpressionContextVisitor(self.provider, ExpressionContext.STORE)
+        )
+        node.body.visit(self)
+        for base in node.bases:
+            base.visit(self)
+        for keyword in node.keywords:
+            keyword.visit(self)
+        for decorator in node.decorators:
+            decorator.visit(self)
+        return False
+
+    def visit_FunctionDef(self, node: cst.FunctionDef) -> Optional[bool]:
```

Comment:
The `returns` of a FunctionDef can also have some names.
Let's also visit it and add a test case for it.
https://libcst.readthedocs.io/en/latest/nodes.html#libcst.FunctionDef